package foodtrail.logic.commands;

import static foodtrail.model.Model.PREDICATE_SHOW_ALL_RESTAURANTS;
import static java.util.Objects.requireNonNull;

import foodtrail.model.Model;

/**
 * Lists all restaurants in the restaurant directory to the user.
 */
public class ListCommand extends Command {

    public static final String COMMAND_WORD = "list";

    public static final String MESSAGE_SUCCESS = "Listed all restaurants";


    @Override
    public CommandResult execute(Model model) {
        requireNonNull(model);
        model.updateFilteredRestaurantList(PREDICATE_SHOW_ALL_RESTAURANTS);
        model.sortRestaurantListByName();
        return new CommandResult(MESSAGE_SUCCESS);
    }
}
